sklearn/feature_selection/mutual_info.py

```diff
@@ -0,0 +1,293 @@
+# Authors: Andrea Bravi <a.bravi@uottawa.ca>
+#          Nikolay Mayorov <n59_ru@hotmail.com>
+# License: 3-clause BSD
+
+from __future__ import division
+
+import numpy as np
+from scipy.sparse import issparse
+from scipy.special import digamma
+
+from ..base import BaseEstimator
+from .base import SelectorMixin
+from ..metrics.cluster.supervised import mutual_info_score
+from ..utils.validation import check_X_y
+from ..neighbors import NearestNeighbors
+from ..preprocessing import scale
+
+
+def _compute_mi_cc(x, y, n_neighbors, scale_variables):
+    """Compute mutual information between two continuous variables.
+
+    Parameter `scale_variables` determine whether to scale variables to
+    the unit variance before computing mutual information. The theoretical
+    value of mutual information doesn't change by that transformation, but
+    in  practice it can be very important to do this scaling. Basically the
+    purpose of this flag is to disable unnecessary scaling in scenarios when it
+    was done in outer scope of this function.
+
+    For the details of the algorithm see references in
```

Comment:
Do you understand intuitively why the formula makes sense? I just see that n_samples_x and n_samples_y should be less if the mutual information should be high. I also see that there is a rigorous math proof but was looking for a straight forward intuition.